lib/agendaitem-order.js

```diff
@@ -0,0 +1,95 @@
+import { query, update, sparqlEscapeUri, sparqlEscapeInt } from 'mu';
+import { reduceResultSet } from './utils';
+
+async function getRelatedAgendaitems(agenda, agendaitemType) {
+  const queryString = `PREFIX dct: <http://purl.org/dc/terms/>
+PREFIX mandaat: <http://data.vlaanderen.be/ns/mandaat#>
+PREFIX schema: <http://schema.org/>
+PREFIX prov: <http://www.w3.org/ns/prov#>
+PREFIX besluitvorming: <https://data.vlaanderen.be/ns/besluitvorming#>
+PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+
+SELECT DISTINCT ?agendaitem ?agendaitemCreated ?agendaitemPosition ?mandateePriority
+WHERE {
+  ${sparqlEscapeUri(agenda)} dct:hasPart ?agendaitem .
+  ?agendaitem schema:position ?agendaitemPosition ;
+              dct:created ?agendaitemCreated ;
+              dct:type ${sparqlEscapeUri(agendaitemType)} .
+  FILTER NOT EXISTS {
+    ?agendaitem prov:wasRevisionOf ?olderAgendaitem .
+  }
+  ?agendaActivity besluitvorming:genereertAgendapunt ?agendaitem ;
+    besluitvorming:vindtPlaatsTijdens ?subcase .
+  OPTIONAL {
+    ?subcase ext:heeftBevoegde ?mandatee .
+    ?mandatee mandaat:rangorde ?mandateePriority .
+  }
+}`;
+  return await query(queryString);
+}
+
+async function applyAgendaitemsOrder(orders) {
+  const queryString = `PREFIX schema: <http://schema.org/>
+
+DELETE {
+  ?agendaitem schema:position ?oldPosition .
+}
+INSERT {
+  ?agendaitem schema:position ?newPosition .
+}
+WHERE {
+  VALUES (?agendaitem ?oldPosition ?newPosition) {
+    ${orders
+      .map(
+        ({ uri, oldPosition, newPosition }) =>
+          `(${sparqlEscapeUri(uri)} ${sparqlEscapeInt(oldPosition)} ${sparqlEscapeInt(newPosition)})`)
+      .join('\n    ')}
+  }
+}`;
+  await update(queryString);
+}
+
+async function reorderAgendaitems(agenda, agendaitemType) {
+  const agendaitems = reduceResultSet(await getRelatedAgendaitems(agenda, agendaitemType));
+
+  // We sort on the mandatee priority. To do this, we simply cast the array of
+  // priorities to a string: [1, 2, 3] → "1,2,3" and we just use string
+  // comparisons to ensure we have a lexicographical sort. E.g. given the
+  // following list of priorities: [ [1, 2], [1], [2, 4], [3], [2, 3] ]
+  // the sorted list will be: [ [1], [1, 2], [2, 3], [2, 4], [3] ]
+  //
+  // If multiple items have the same priority list, we sort them based on their
+  // existing position. This way manual reorders inside a priority group are
+  // kept when submitting a new agendaitem.
+  const sortedAgendaitems = agendaitems.sort(
+    (a1, a2) => {
+      const priority1 = Array.isArray(a1.mandateePriority) ? a1.mandateePriority.sort().toString() : String(a1.mandateePriority);
+      const priority2 = Array.isArray(a2.mandateePriority) ? a2.mandateePriority.sort().toString() : String(a2.mandateePriority);
+      return priority1 === priority2
+        ? a1.agendaitemPosition - a2.agendaitemPosition
+        : priority1 < priority2
+          ? -1
+          : 1
+    });
```

Comment:
Does the sort on numbering work with 11 mandatees?
f.e. the group [1, 2]  is more important than the group [1, 10, 11]
That is issue we encountered in newsletter service and to solve that we used letters instead of numbers.

Reply:
Dealt with here https://github.com/kanselarij-vlaanderen/agenda-submission-service/pull/4/commits/9a67f98e01bc2e7879a28baa51b66e20e88ab507 and here https://github.com/kanselarij-vlaanderen/agenda-submission-service/pull/4/commits/e22fa31f07015d133c2c6dfcdc8ca96afb5b8f42

Not sure how often (if at all?) there will be more than 9 ministers, but I went for a more "general" approach. Instead of replacing numbers with letters (which would still limit us to 26 ministers), the priority number is converted to a string which is then padded with zeroes according to the longest number that was encountered.
So `[[1, 2], [1, 10, 11]]` will get sorted as `[['01', '02'], ['01', '10', '11']]`

Reply:
From what I remember, there can only be max 11 ministers.
Direct qoute from VP website:
"De Vlaamse Regering bestaat uit maximaal elf ministers. De huidige Vlaamse Regering telt negen ministers"